src/widgets/InfoBadge.js

```diff
@@ -1,129 +1,103 @@
-import React from 'react';
+/* eslint-disable react/forbid-prop-types */
+import React, { useState, useCallback } from 'react';
 import PropTypes from 'prop-types';
-import { StyleSheet, View, TouchableHighlight, Text } from 'react-native';
+import { View, TouchableHighlight, Text } from 'react-native';
 import Popover from 'react-native-popover-view';
-import Badge from './Badge';
+
 import { getBadgeData } from '../utilities/getBadgeData';
-import { SUSSOL_ORANGE } from '../globalStyles';
+import { usePopover } from '../hooks';
+
+import Badge from './Badge';
 
-export class InfoBadge extends React.PureComponent {
-  state = {
-    isPopOverVisible: false,
-    rect: {},
-  };
+import { SUSSOL_ORANGE } from '../globalStyles';
 
-  showPopover() {
-    // Open popover and determine where to open it on canvas
-    this.touchableContainer.measure((ox, oy, width, height, px, py) => {
-      this.setState({
-        isPopOverVisible: true,
-        rect: { x: px, y: py, width, height },
-      });
-    });
-  }
+const animationConfig = { duration: 150 };
 
-  closePopover() {
-    this.setState({ isPopOverVisible: false });
-  }
+export const InfoBadge = ({
+  popoverBackgroundStyle,
+  children,
+  mainWrapperStyle,
+  popoverPosition,
+  routeName,
+  arrowStyle,
+  popoverStyle,
+  badgeTextStyle,
+  touchableContainerStyle,
+}) => {
+  const [ref] = useState(React.createRef());
+  const [visible, show, close] = usePopover(ref);
 
-  render() {
-    const { children, mainWrapperStyle, popoverPosition, routeName } = this.props;
-    const { isPopOverVisible, rect } = this.state;
-    const info = getBadgeData(routeName);
+  // Get total of all the count variables in the info array. We want to show it on the badge
+  const info = getBadgeData(routeName);
+  const pendingCount = info.reduce((total, item) => total + (item.count || 0), 0);
 
-    // Get total of all the count variables in the info array. We want to show it on the badge
-    const pendingCount = info.reduce((total, item) => total + (item.count || 0), 0);
+  // show 99+ if the number is greater then 99 to limit the number of characters.
+  const unfinalisedCountText = pendingCount > 99 ? '99+' : pendingCount;
 
-    // Since having multiple digit text can break the badge, we are showing 99+
-    // if the number is greater then 99.
-    const unfinalisedCountText = pendingCount > 99 ? '99+' : pendingCount;
+  const popoverText = useCallback(
+    (item, key) =>
+      (item.count > 0 || item.text) && (
+        <Text style={badgeTextStyle} key={key}>
+          {`${item.title}: ${item.count || item.text}`}
+        </Text>
+      ),
+    []
+  );
```

Comment:
@joshxg I just ran the code in emulator and I got this error.

![Screenshot 2019-11-07 at 11 52 45](https://user-images.githubusercontent.com/683452/68364213-64033080-0155-11ea-84eb-dcb848d52e3f.jpg)

I had encountered this issue before when I used `view` as wripper. You might notice the `text` wrapper rather then `view` wrapper in the previous code. 

You did not get this error at your end?